src/lib/y2network/widgets/s390_common.rb

```diff
@@ -0,0 +1,331 @@
+# find current contact information at www.suse.com.
+
+require "cwm/common_widgets"
+require "cwm/custom_widget"
+
+module Y2Network
+  module Widgets
+    # Widget for setting the s390 device write channel
+    class S390LanCmdTimeout < CWM::InputField
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        _("&LANCMD Time-Out")
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = @settings.timeout
+      end
+
+      # @see CWM::AbstractWidget
+      def store
+        @settings.timeout = value
+      end
+
+      # @see CWM::AbstractWidget
+      def help
+        _("<p>Specify the <b>LANCMD Time-Out</b> for this interface.</p>")
+      end
+    end
+
+    # Widget for setting the Ctc device protocol to be used
+    class S390Protocol < CWM::ComboBox
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = @settings.protocol.to_s
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        _("&Protocol")
+      end
+
+      # @see CWM::AbstractWidget
+      def items
+        [
+          # ComboBox item: CTC device protocol
+          ["0", _("Compatibility Mode")],
+          # ComboBox item: CTC device protocol
+          ["1", _("Extended Mode")],
+          # ComboBox item: CTC device protocol
+          ["2", _("CTC-Based tty (Linux to Linux Connections)")],
+          # ComboBox item: CTC device protocol
+          ["3", _("Compatibility Mode with OS/390 and z/OS")]
+        ]
+      end
+
+      # @see CWM::AbstractWidget
+      def store
+        @settings.protocol = value.to_i
+      end
+
+      # @see CWM::AbstractWidget
+      def help
+        _("<p>Choose the <b>Protocol</b> for this interface.</p>")
+      end
+    end
+
+    # Widget for specifying whether use the port number 0 or 1
+    class S390PortNumber < CWM::ComboBox
+      # Constructor
+      #
+      # @param settings [Y2Network::InterfaceConfigBuilder]
+      def initialize(settings)
+        textdomain "network"
+
+        @settings = settings
+      end
+
+      # @see CWM::AbstractWidget
+      def init
+        self.value = @settings.port_number.to_s
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        _("Port Number")
+      end
+
+      # @see CWM::AbstractWidget
+      def items
+        [["0", "0"], ["1", "1"]]
+      end
+
+      # @see CWM::AbstractWidget
+      def store
+        @settings.port_number = value.to_i
+      end
+
+      # @see CWM::AbstractWidget
+      def help
+        _("<p>Choose the <b>Port Number</b> for this interface.</p>")
```

Comment:
hmm, is this help really helpful? I would prefer to have here something that describe why you should pick port number andwhat it is

Reply:
Them really looks like saying the same the label already describe xD, so, will check all the help texts trying to make them more helpful.

Reply:
Although from other POV, I suspect most of the users that activate an S390 group device do not need the help at all ... :P